### PR TITLE
Add jam for windows to releases

### LINKED ISSUE #46: [Request] Windows release artifacts
`jam` would be a useful tool to have for Windows users in addition to Linux and Mac users. Would you consider adding a Windows build on your next release?

### PR DESCRIPTION
**Pull request by ameyer-pivotal**
Resolves #46 

Signed-off-by: Andrew Meyer <meyeran@vmware.com>

### CODE REVIEW

**Comment by ameyer-pivotal** [comment]
Went ahead and PR'd in my original request #46. That said, I haven't fully validated jam for Windows. Are there any inherent incompatibilities that led to excluding Windows in the first place? Thanks!

File: scripts/package.sh [hunk 1]
```diff
@@ -16,6 +16,8 @@ function main() {
       GOOS="${os}" GOARCH="amd64" go build -o "${ARTIFACTS_DIR}/jam-${os}" ./cargo/jam/main.go
       chmod +x "${ARTIFACTS_DIR}/jam-${os}"
     done
+    echo "* building jam on windows"
+    GOOS="windows" go build -o "${ARTIFACTS_DIR}/jam-windows.exe" ./cargo/jam/main.go
```
Comment (ameyer-pivotal) [inline scripts/package.sh:20]: Chose not to roll this into the for loop due to the file extension. This is a quick, simple solution, but happy to clean it up if needed :)

File: fs/copy.go [hunk 2]
```diff
@@ -63,7 +63,7 @@ func copyFile(source, destination string) error {
 		return err
 	}
 
-	err = destinationFile.Chmod(info.Mode())
+	err = os.Chmod(destination, info.Mode())
```
Comment (ameyer-pivotal) [inline fs/copy.go:66]: `File.Chmod` causes a syscall error for Windows. See related issue: https://github.com/google/skicka/issues/77#issuecomment-97988722

**Comment by ryanmoran** [comment]
This looks great. I don't know of anything off the top of my head that would make `jam` incompatible with Windows, but I've also not tried it.

One note, the PR should also update the workflow to create a draft release: https://github.com/paketo-buildpacks/packit/blob/8dfde6003401202f4520db53b78eacd81e496b56/.github/workflows/create-release.yml#L40-L62. In that section, you will need to include the Windows `jam` executable so that it gets included as an asset in the release.

**Comment by ameyer-pivotal** [comment]
> This looks great. I don't know of anything off the top of my head that would make `jam` incompatible with Windows, but I've also not tried it.
> 
> One note, the PR should also update the workflow to create a draft release:
> 
> https://github.com/paketo-buildpacks/packit/blob/8dfde6003401202f4520db53b78eacd81e496b56/.github/workflows/create-release.yml#L40-L62
> 
> In that section, you will need to include the Windows `jam` executable so that it gets included as an asset in the release.

Thanks @ryanmoran. I had already included a change to the release workflow. Is there something I'm still missing?

File: .github/workflows/create-release.yml [hunk 3]
```diff
@@ -58,6 +58,11 @@ jobs:
               "path": "build/jam-linux",
               "name": "jam-linux",
               "content_type": "application/octet-stream"
+            },
```
Comment (ameyer-pivotal) [inline .github/workflows/create-release.yml:61]: @ryanmoran See here for adding to release artifacts

**Comment by ryanmoran** [comment]
No. I just failed to fully read the file names. 😞

**Pull request merged**